From d67b07e06a5feab131bcea8ef51f01b2c17e3a77 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Sat, 31 May 2025 18:47:43 +0300
Subject: [PATCH] [JSW-808] jgs: enable palad build

With this patch, added required config
files to enable palad build.

Change-Id: I4a4706b1e884c73abedaa9161ecd7abbd98a1e61
---

diff --git a/boards/intel/jgs/Kconfig.jgs_mgmt_palad b/boards/intel/jgs/Kconfig.jgs_mgmt_palad
new file mode 100644
index 0000000..ee61dda
--- /dev/null
+++ b/boards/intel/jgs/Kconfig.jgs_mgmt_palad
@@ -0,0 +1,5 @@
+# Copyright 2022 HabanaLabs, Ltd. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_JGS_MGMT_PALAD
+	select SOC_JGS_MGMT
diff --git a/boards/intel/jgs/Kconfig.jgs_preboot_palad b/boards/intel/jgs/Kconfig.jgs_preboot_palad
new file mode 100644
index 0000000..9e94aa2
--- /dev/null
+++ b/boards/intel/jgs/Kconfig.jgs_preboot_palad
@@ -0,0 +1,5 @@
+# Copyright 2025 HabanaLabs, Ltd. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_JGS_PREBOOT_PALAD
+	select SOC_JGS_PREBOOT
diff --git a/boards/intel/jgs/Kconfig.jgs_rst_palad b/boards/intel/jgs/Kconfig.jgs_rst_palad
new file mode 100644
index 0000000..5d941bc
--- /dev/null
+++ b/boards/intel/jgs/Kconfig.jgs_rst_palad
@@ -0,0 +1,5 @@
+# Copyright 2022 HabanaLabs, Ltd. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_JGS_RST_PALAD
+	select SOC_JGS_RST
diff --git a/boards/intel/jgs/board.yml b/boards/intel/jgs/board.yml
index 4c7d9dc..045554e 100644
--- a/boards/intel/jgs/board.yml
+++ b/boards/intel/jgs/board.yml
@@ -23,9 +23,15 @@
     vendor: intel
     socs:
       - name: rst
-
-
-
-
-
-
+  - name: jgs_preboot_palad
+    vendor: intel
+    socs:
+      - name: preboot
+  - name: jgs_rst_palad
+    vendor: intel
+    socs:
+      - name: rst
+  - name: jgs_mgmt_palad
+    vendor: intel
+    socs:
+      - name: mgmt
diff --git a/boards/intel/jgs/jgs_mgmt_palad.dts b/boards/intel/jgs/jgs_mgmt_palad.dts
new file mode 100644
index 0000000..3022009
--- /dev/null
+++ b/boards/intel/jgs/jgs_mgmt_palad.dts
@@ -0,0 +1,19 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/dts-v1/;
+
+#include <jgs_io_mgmt.dtsi>
+
+&refclk {
+	clock-frequency = <24100>;
+};
+
+&uart0 {
+	current-speed = <300>;
+	status = "okay";
+};
diff --git a/boards/intel/jgs/jgs_mgmt_palad_defconfig b/boards/intel/jgs/jgs_mgmt_palad_defconfig
new file mode 100644
index 0000000..70b3274
--- /dev/null
+++ b/boards/intel/jgs/jgs_mgmt_palad_defconfig
@@ -0,0 +1,23 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CPU_ARCHS_NO_HW_ATOMIC=y
+CONFIG_EXCEPTION_DEBUG=y
+CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE=2048
+CONFIG_BUILD_OUTPUT_BIN=y
+CONFIG_MAIN_STACK_SIZE=2048
+CONFIG_PRINTK=y
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_LOG_MAX_LEVEL=4
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_EMPTY_WEAK_FUNCTIONS=y
+CONFIG_LINKER_ORPHAN_SECTION_PLACE=y
+CONFIG_XIP=n
+#CONFIG_CLOCK_CONTROL=y
+#CONFIG_CLOCK_CONTROL_HL_PLL=y
+#CONFIG_CLOCK_CONTROL_HL_PLL_V2=y
+CONFIG_HEAP_MEM_POOL_SIZE=4096
+CONFIG_HL_EFUSE=y
diff --git a/boards/intel/jgs/jgs_preboot_palad.dts b/boards/intel/jgs/jgs_preboot_palad.dts
new file mode 100644
index 0000000..26dd460
--- /dev/null
+++ b/boards/intel/jgs/jgs_preboot_palad.dts
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2023 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/dts-v1/;
+
+#include <jgs_preboot.dtsi>
+
+/ {
+	model = "jgs_preboot";
+};
+
+&refclk {
+	/* clock values differ between images on pldm */
+	clock-frequency = <21300>;
+};
+
+&uart0 {
+	current-speed = <300>;
+	status = "okay";
+};
diff --git a/boards/intel/jgs/jgs_preboot_palad_defconfig b/boards/intel/jgs/jgs_preboot_palad_defconfig
new file mode 100644
index 0000000..d03c254
--- /dev/null
+++ b/boards/intel/jgs/jgs_preboot_palad_defconfig
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CPU_ARCHS_NO_HW_ATOMIC=y
+CONFIG_EXCEPTION_DEBUG=y
+CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE=2048
+CONFIG_HEAP_MEM_POOL_SIZE=3072
+CONFIG_BUILD_OUTPUT_BIN=y
+CONFIG_MAIN_STACK_SIZE=5120
+CONFIG_PRINTK=y
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_LOG_MAX_LEVEL=4
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_MBEDTLS=y
+CONFIG_MBEDTLS_BUILTIN=y
+CONFIG_MBEDTLS_SHA384=y
+CONFIG_LIBFDT=y
+CONFIG_LIBBOOTINFO=y
+# boot info addr is spsram base This address is yet to be finalized
+CONFIG_BOOTINFO_ADDR=0x78a00000
+CONFIG_BOOTINFO_SIZE=0x8000
+CONFIG_LIBSOC_INIT=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_EMPTY_WEAK_FUNCTIONS=y
+CONFIG_LINKER_ORPHAN_SECTION_PLACE=y
+CONFIG_MULTITHREADING=y
+CONFIG_FAULT_THREAD_DUMP=y
+CONFIG_HEAP_MEM_POOL_SIZE=3072
+CONFIG_MINIMAL_LIBC_OPTIMIZE_STRING_FOR_SIZE=n
+CONFIG_BOOT_IF_PSEUDO_RO_TIMER=y
+CONFIG_HL_EFUSE=y
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=1000000000
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=10000
+CONFIG_MAIN_THREAD_PRIORITY=4
+CONFIG_ARCV2_INTERRUPT_UNIT=y
+CONFIG_ARCV2_TIMER=y
+CONFIG_POLL=y
+CONFIG_THREAD_NAME=y
+CONFIG_XIP=n
+#CONFIG_ISOLATION_AND_BINNING_INTEL_FS=y
+#CONFIG_PCIE_MBOX=y
+#CONFIG_PCIE_SRIOV=y
+#CONFIG_CLOCK_CONTROL=y
+#CONFIG_CLOCK_CONTROL_HL_PLL=y
+#CONFIG_CLOCK_CONTROL_HL_PLL_V2=y
+#CONFIG_GPU_UTILS=y
+CONFIG_CRC=y
diff --git a/boards/intel/jgs/jgs_rst_palad.dts b/boards/intel/jgs/jgs_rst_palad.dts
new file mode 100644
index 0000000..4f03ebb
--- /dev/null
+++ b/boards/intel/jgs/jgs_rst_palad.dts
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2024 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+/dts-v1/;
+
+#include <jgs_rst.dtsi>
+
+/ {
+	model = "jgs_rst";
+	compatible = "intel,jgs_rst";
+
+	chosen {
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+	};
+};
+
+&refclk {
+	/* clock values differ between images on pldm */
+	clock-frequency = <21300>;
+};
+
+&uart0 {
+	current-speed = <300>;
+	status = "okay";
+};
diff --git a/boards/intel/jgs/jgs_rst_palad_defconfig b/boards/intel/jgs/jgs_rst_palad_defconfig
new file mode 100644
index 0000000..d3d603f
--- /dev/null
+++ b/boards/intel/jgs/jgs_rst_palad_defconfig
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CPU_ARCHS_NO_HW_ATOMIC=y
+CONFIG_EXCEPTION_DEBUG=y
+CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE=2048
+CONFIG_HEAP_MEM_POOL_SIZE=1280
+CONFIG_BUILD_OUTPUT_BIN=y
+CONFIG_MAIN_STACK_SIZE=5120
+CONFIG_PRINTK=y
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
+CONFIG_LOG_MAX_LEVEL=4
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_EMPTY_WEAK_FUNCTIONS=y
+CONFIG_LINKER_ORPHAN_SECTION_PLACE=y
+CONFIG_MINIMAL_LIBC_OPTIMIZE_STRING_FOR_SIZE=n
+CONFIG_BOOT_IF_PSEUDO_RO_TIMER=y
+CONFIG_XIP=n
diff --git a/drivers/misc/common_utils/gpu/jgs/palad.c b/drivers/misc/common_utils/gpu/jgs/palad.c
index 5039793..a649e87 100644
--- a/drivers/misc/common_utils/gpu/jgs/palad.c
+++ b/drivers/misc/common_utils/gpu/jgs/palad.c
@@ -8,8 +8,8 @@
 #include <efuse.h>
 #include <hl_devs.h>
 #include <zephyr/logging/log.h>
-#include <fs1_fw_if.h>
-#include <fs1_efuse.h>
+#include <jgs_fw_if.h>
+#include <jgs_efuse.h>
 
 LOG_MODULE_REGISTER(fs1_palad);
 
